### PR TITLE
Implement usage of scopes in pool importing of browser providers. Fixes #2221

### LINKED ISSUE #2221: Testcafe doesn't work with privately scoped providers
### Are you requesting a feature or reporting a bug?
Bug

### What is the current behavior?
When publishing a browser provider to a private npm repository under a scope (@bsmithb2 for example) the installed provider fails to be detected by the browser provider pool. 

### What is the expected behavior?
That you can install the browser provider from a private npm repo in a scope and the browser provider is detected. 

### How would you reproduce the current behavior (if this is a bug)?
Create a demo browser provider. Publish to a private repo under a scope. Install the private, scoped npm package. Run testcafe -b. Note the provider is not detected. 

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:

Test code

```js

```

### Specify your

* operating system: Windows
* testcafe version: 0.18.6
* node.js version: 9.8.0

### PR DESCRIPTION
**Pull request by bsmithb2**

### CODE REVIEW

File: src/browser/provider/module-loader.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const BROWSER_PROVIDER_SCOPED_PACKAGE_RE = /^@\S+\/testcafe-browser-provider\S/;
+
+export default class BrowserProviderModuleLoader {
+    loadModule (providerName) {
+        var providerObject = null;
+
+        if (this.isPrivateModule(providerName))
+            providerObject = require(`${providerName}`);
```
Comment (AndreyBelym) [inline src/browser/provider/module-loader.js]: I think it's better to extract `scope` from RegExp and use
```
providerObject = require(`${scope}/testcafe-browser-provider-${providerName}`);
```
here

**Comment by bsmithb2** [comment]
So you mean the arg passed in at the command line will be testcafe @scope/tail-end-of-package-name test.js (ie the one provided by the yeoman wizard) rather than testcafe @scope/testcafe-browser-provider-tail-end-of-package-name year.js?

Thanks!

**Comment by AndreyBelym** [comment]
Yes, I think it's more convenient to write `@scope/myprovider` instead of `@sccope/testcafe-browser-provider-myprovider. I think it's good to have naming convention, because we will be able to distinguish browser providers from all other NPM packages just by looking at the name.

**Comment by AndreyBelym** [comment]
I can do it by myself along with rebasing, if you don't have time to do it. Just let me know, please.

**Comment by bsmithb2** [comment]
Nah I’m happy to finish the job. Enjoying my first contribution to open source!

**Comment by bsmithb2** [comment]
@AndreyBelym - when I run gulp-test client on both windows and linux, the build stops at "QUnit server listens on http://localhost:2000"

Any ideas what I've done wrong?

File: src/browser/provider/module-loader.js (hunk 1 above)
Comment (bsmithb2) [reply src/browser/provider/module-loader.js]: Updated - please have a look @AndreyBelym and let me know any feedback.  Note that I've passed all the test-server tests but on my machine (windows 10 XPS 15, and on my linux VM) I can't run test-functional-local and test-client... the functional tests break in weird ways that aren't replicable in the build process, and the test-client never start after booting the qunit server on port 2000?

**Pull request closed without merging**